### PR TITLE
Permutation method names and assertions

### PR DESCRIPTION
**Pull request by axtimwalde**
changed method name to Views.permuteCoordinates because
Views.permute already permutes axes, single axis and all axes can
share the same name with different parameters

added assertions for input sanity check (bijectivity and matching
dimensions)

### CODE REVIEW

File: src/main/java/net/imglib2/transform/integer/permutation/AbstractPermutationTransform.java [hunk 1]
```diff
@@ -7,6 +7,10 @@
 
 /**
  * Bijective integer transform mapping between integer coordinates in [0,n-1].
+ * Currently, this transform handles only coordinates in the integer range
+ * because it is implemented using primitive arrays for efficiency reasons whose
+ * index is and integer. Expect this permutation to be transferred to long
```
Comment (hanslovsky) [inline src/main/java/net/imglib2/transform/integer/permutation/AbstractPermutationTransform.java]: Missing word:
```
[...]for efficiency reasons whose index is and integer
```
Rewrite to:
```
[...]is implemented using primitive arrays with integer indices for efficiency reasons.
```
?

File: src/main/java/net/imglib2/transform/integer/permutation/PermutationTransform.java [hunk 2]
```diff
@@ -107,4 +108,23 @@ public PermutationTransform inverse()
...
+	/**
+	 * Test whether a given lut can be applied to an interval. This means that
+	 * the interval is a hypercube at min = 0<sup><em>n</em></sup> with size =
+	 * lut.length.
+	 *
+	 * @param interval
+	 * @param lut
+	 * @return
+	 */
+	static public boolean checkInterval( final Interval interval, final int[] lut )
```
Comment (hanslovsky) [inline src/main/java/net/imglib2/transform/integer/permutation/PermutationTransform.java]: add `AbstractPermutationTransform.checkInterval( Interval, int[], int )` and then do:
```
checkInterval( Interval, int[] )
{
    for ( d in Interval.numDimensions() )
        checkInterval( Interval, int[], int )
}
```
?

File: src/main/java/net/imglib2/view/Views.java [hunk 3]
```diff
@@ -1124,112 +1125,114 @@ public static boolean isZeroMin( final Interval interval )
...
-	 * RandomAccessibleInterval, i.e. for dimension d of a target coordinate:
-	 * targetCoordinate[ d ] = permutation[ sourceCoordinate[ d ] ] Sanity check
-	 * of the inputs is up to the user.
+	 * Inverse bijective permutation of the integer coordinates of one dimension
+	 * of a {@link RandomAccessibleInterval}.
 	 *
 	 * @param source
-	 *            must be an n-dimensional hypercube with dimension d being of
-	 *            the same size as the permutation array
+	 *            must be an <em>n</em>-dimensional with dimension(dimension) ==
```
Comment (hanslovsky) [inline src/main/java/net/imglib2/view/Views.java]: word missing:
```
[...]must be an <em>n</em>-dimensional <hypercube?> with[...]
```
and `dimension(dimension)` -> `dimension(d)` ?

Comment (axtimwalde) [inline src/main/java/net/imglib2/view/Views.java]: Nope, no cube here.  Fixing...

File: src/main/java/net/imglib2/transform/integer/permutation/PermutationTransform.java (hunk 2 above)
Comment (axtimwalde) [inline src/main/java/net/imglib2/transform/integer/permutation/PermutationTransform.java]: Possible but not crucial.  It's assertion code after all, never executed.

**Pull request merged**